src/overlays/actors/ovl_Bg_Bombwall/z_bg_bombwall.c

```diff
@@ -36,29 +61,192 @@ const ActorInit Bg_Bombwall_InitVars = {
     (ActorFunc)BgBombwall_Update,
     (ActorFunc)BgBombwall_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bombwall/func_8086E7D0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bombwall/func_8086E850.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bombwall/BgBombwall_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bombwall/func_8086EAC0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bombwall/BgBombwall_Destroy.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bombwall/func_8086EB5C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bombwall/func_8086ED50.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bombwall/func_8086ED70.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bombwall/func_8086EDFC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bombwall/func_8086EE40.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bombwall/func_8086EE94.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bombwall/BgBombwall_Update.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(uncullZoneForward, 1800, ICHAIN_CONTINUE), // unk f4
+    ICHAIN_F32(uncullZoneScale, 300, ICHAIN_CONTINUE),    // unk f8
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),    // unk fc
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bombwall/BgBombwall_Draw.s")
+Vec3s D_8086F010[] = { { 0x0028, 0x0055, 0x0015 }, { 0xFFD5, 0x006B, 0x000E }, { 0xFFFF, 0x008E, 0x000E },
+                       { 0xFFE5, 0x002C, 0x001B }, { 0x001C, 0x0018, 0x0014 }, { 0xFFD9, 0x0036, 0x0015 },
+                       { 0x0031, 0x0032, 0x0014 } };
+
+void BgBombwall_InitDynapoly(BgBombwall* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 pad2;
+    ColHeader* colHeader;
+
+    colHeader = NULL;
+    DynaPolyInfo_SetActorMove(&this->dyna, 0);
+    DynaPolyInfo_Alloc(&D_050041B0, &colHeader);
+    this->dyna.dynaPolyId =
+        DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, colHeader);
+
+    if (this->dyna.dynaPolyId == 0x32) {
+        // Warning : move BG login failed
+        osSyncPrintf((const char*)"Warning : move BG 登録失敗(%s %d)(arg_data 0x%04x)\n", "../z_bg_bombwall.c", 0xF3,
+                     this->dyna.actor.params);
```

Comment:
you can remove (const char*)
also the 0xF3 is a line number so it can be decimal